### PR TITLE
[GP-6] Trash handling for shows

### PR DESCRIPTION
**Pull request by andrasguseo**
I did this for fun. Consider it a community contribution. :)

[GP-6]

Origin:
https://wordpress.org/support/topic/gig-press-viewing-trash/

[GP-6]: https://moderntribe.atlassian.net/browse/GP-6

### CODE REVIEW

**Comment by andrasguseo** [comment]
I originally branched it from master. Then I realized the latest is G20.04 which has not been merged into master.
I don't know what happened did, but I guess that commit 5978986 moved the branch over to G20.04
If this created only a bigger mess, then sorry. :)

File: admin/handlers.php [hunk 1]
```diff
@@ -1006,6 +1006,28 @@ function gigpress_empty_trash() {
...
+// ======================
+
+function gigpress_restore_show() {
+
+	global $wpdb;
+	$wpdb->show_errors();
+	check_admin_referer('gigpress-action');
+
+	$show = [ 'show_status' => 'active' ];
+	$where = array('show_id' => $_GET['show_id']);
```
Comment (sc0ttkclark) [inline admin/handlers.php:1019]:
```suggestion
	$where = array( 'show_id' => absint( $_GET['show_id'] ) );
```

File: admin/handlers.php [hunk 2]
```diff
@@ -1006,6 +1006,28 @@ function gigpress_empty_trash() {
...
+
+	$restore_show = $wpdb->update(GIGPRESS_SHOWS, $show, $where);
```
Comment (sc0ttkclark) [inline admin/handlers.php:1021]:
```suggestion
	$restore_show = $wpdb->update( GIGPRESS_SHOWS, $show, $where );
```

File: admin/handlers.php [hunk 3]
```diff
@@ -1006,6 +1006,28 @@ function gigpress_empty_trash() {
...
+
+	if( $restore_show ) { ?>
+		<div id="message" class="updated fade"><p><?php _e("The selected show has been restored.", "gigpress"); ?></p></div>
```
Comment (sc0ttkclark) [inline admin/handlers.php:1024]:
```suggestion
		<div id="message" class="updated fade"><p><?php esc_html_e( 'The selected show has been restored.', 'gigpress' ); ?></p></div>
```

File: admin/handlers.php [hunk 4]
```diff
@@ -1006,6 +1006,28 @@ function gigpress_empty_trash() {
...
+	<?php } else { ?>
+		<div id="message" class="error fade"><p><?php _e("We ran into some trouble restoring the show. Sorry.", "gigpress"); ?></p></div>
```
Comment (sc0ttkclark) [inline admin/handlers.php:1026]:
```suggestion
		<div id="message" class="error fade"><p><?php esc_html_e( 'We ran into some trouble restoring the show. The selected show was not restored.', 'gigpress' ); ?></p></div>
```

**Comment by andrasguseo** [comment]
@sc0ttkclark just saying, it has a ticket
https://moderntribe.atlassian.net/browse/GP-6

Or do you need something else?

**Comment by sc0ttkclark** [comment]
All good, updated PR accordingly

**Comment by sc0ttkclark** [comment]
This was merged too soon, so I've removed these commits from `release/G20.07`, we can open a new PR for this

**Comment by sc0ttkclark** [comment]
https://github.com/moderntribe/gigpress/pull/56

**Pull request merged**